ulib/LowStar.RVector.fst

```diff
@@ -0,0 +1,1060 @@
+module LowStar.RVector
+
+open FStar.All
+open FStar.Integers
+open FStar.Classical
+open LowStar.Modifies
+open LowStar.Regional
+open LowStar.Vector
+
+module HH = FStar.Monotonic.HyperHeap
+module HS = FStar.HyperStack
+module HST = FStar.HyperStack.ST
+module MHS = FStar.Monotonic.HyperStack
+module S = FStar.Seq
+module B = LowStar.Buffer
+module V = LowStar.Vector
+
+module U32 = FStar.UInt32
+
+/// Utilities
+
+// `HST.new_region` does not guarantee `modifies loc_none h0 h1`, which is
+// a bit annoying. It can be proven by `B.modifies_none_modifies` and here
+// `new_region_` is exactly doing that.
+val new_region_:
+  r0:HH.rid -> 
+  HST.ST HH.rid
+    (fun _ -> HST.is_eternal_region r0)
+    (fun h0 r1 h1 ->
+      MHS.fresh_region r1 h0 h1 /\
+      HH.extends r1 r0 /\
+      MHS.get_hmap h1 == Map.upd (MHS.get_hmap h0) r1 Monotonic.Heap.emp /\
+      HH.color r1 = HH.color r0 /\
+      HyperStack.ST.is_eternal_region r1 /\
+      modifies loc_none h0 h1)
+let new_region_ r0 =
+  let hh0 = HST.get () in
+  let r1 = HST.new_region r0 in
+  let hh1 = HST.get () in
+  B.modifies_none_modifies hh0 hh1;
+  r1
```

Comment:
Tahina thinks that `new_region_modifies` from `LowStar.Monotonic.Buffer` should trigger. Do you know why you've had to define a separate helper?